test/spec/modules/greenbidsBidAdapter_specs.js

```diff
@@ -0,0 +1,964 @@
+        platform: {
+          brand: 'macOS',
+          version: ['12', '4', '0']
+        },
+        browsers: [
+          {
+            brand: 'Chromium',
+            version: ['106', '0', '5249', '119']
+          },
+          {
+            brand: 'Google Chrome',
+            version: ['106', '0', '5249', '119']
+          },
+          {
+            brand: 'Not;A=Brand',
+            version: ['99', '0', '0', '0']
+          }
+        ],
+        mobile: 0,
+        model: '',
+        bitness: '64',
+        architecture: 'x86'
+      }
+      );
+
+      const defaultRequest = spec.buildRequests(bidRequests, bidderRequestDefault);
+      expect(JSON.parse(defaultRequest.data).userAgentClientHints).to.not.exist;
+    });
+
+    it('should use good mediaTypes banner sizes', function () {
+      const mediaTypesBannerSize = {
+        'mediaTypes': {
+          'banner': {
+            'sizes': [300, 250]
+          }
+        }
+      };
+      checkMediaTypesSizes(mediaTypesBannerSize, '300x250');
+    });
+  });
+
+  describe('Global Placement Id', function () {
+    let bidRequests = [
+      {
+        'bidder': 'greenbids',
+        'params': {
+        },
+        'adUnitCode': 'adunit-code-1',
+        'sizes': [[300, 250], [300, 600]],
+        'bidId': '30b31c1838de1e',
+        'bidderRequestId': '22edbae2733bf6',
+        'auctionId': '1d1a030790a475',
+        'creativeId': 'er2ee',
+        'deviceWidth': 1680
+      },
+      {
+        'bidder': 'greenbids',
+        'params': {
+        },
+        'adUnitCode': 'adunit-code-2',
+        'sizes': [[300, 250], [300, 600]],
+        'bidId': '30b31c1838de1f',
+        'bidderRequestId': '22edbae2733bf6',
+        'auctionId': '1d1a030790a475',
+        'creativeId': 'er2ef',
+        'deviceWidth': 1680
+      }
+    ];
+
+    it('should add gpid if ortb2Imp.ext.gpid is present and is non empty', function () {
+      const updatedBidRequests = bidRequests.map(function (bidRequest, index) {
+        return {
+          ...bidRequest,
+          ortb2Imp: {
+            ext: {
+              gpid: '1111/home-left-' + index
+            }
+          }
+        };
+      }
+      );
+      const request = spec.buildRequests(updatedBidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.data[0].gpid).to.equal('1111/home-left-0');
+      expect(payload.data[1].gpid).to.equal('1111/home-left-1');
+    });
+
+    it('should not add gpid if ortb2Imp.ext.gpid is present but empty', function () {
+      const updatedBidRequests = bidRequests.map(bidRequest => ({
+        ...bidRequest,
+        ortb2Imp: {
+          ext: {
+            gpid: ''
+          }
+        }
+      }));
+
+      const request = spec.buildRequests(updatedBidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      return payload.data.forEach(bid => {
+        expect(bid).not.to.have.property('gpid');
+      });
```

Comment:
question: why do you need a `return` here while you don't have any in other tests? (Moreover, I believe `forEach` returns nothing so...).